cpp/src/cider/exec/nextgen/function/CiderStringFunctions.cpp

```diff
@@ -140,48 +136,72 @@ const uint8_t ascii_char_lower_map[] = {
     224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239,
     240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252, 253, 254, 255};
 
+ALWAYS_INLINE void do_lower(char* out, const char* in, int str_len) {
```

Comment:
label all pointers as ```__restrict``` to eliminate overlap check code, I think there is no memory overlap in ```query_func```.